plugins/missions.js

```diff
@@ -190,17 +194,137 @@ window.plugin.missions = {
 		}
 	},
 
-	showMissionListDialog: function(missions) {
-		dialog({
-			html: this.renderMissionList(missions),
-			height: 'auto',
-			width: '400px',
-			collapseCallback: this.collapseFix,
-			expandCallback: this.collapseFix,
-		}).dialog('option', 'buttons', {
-			'Create new mission': function() { open('//missions.ingress.com'); },
-			'OK': function() { $(this).dialog('close'); },
-		});
+	showMissionListDialog: function(missions, caption, isPortalList) {
+
+		this.isShowingPortalList = isPortalList;
+
+		// Check whether dialog is already open
+		let openDialog = window.DIALOGS['dialog-missionsList'];
+		if(openDialog)
+		{
+			// Dialog already there, expand if collpased
+			// NOTE: It would be really cool if the IITC core had some function to read and modify collpased state of dialogs
+			// but it hasn't so we need to access internal information here
+			if(plugin.missions.isMissionListCollapsed)
+			{
+				let dia   = $(openDialog).closest('.ui-dialog');
+				let button   = dia.find('.ui-dialog-titlebar-button-collapse');
+				if(button)
+				{
+					$(button).click();
+				}
+			}
+		}
+		else
+		{
+			// If dialog is not open, open it
+			dialog({
+				id: 'missionsList',
+				html: this.renderMissionList(missions),
+				height: 'auto',
+				width: '400px',
+				collapseCallback: this.onCollapseMissionList,
+				expandCallback: this.onExpandMissionList,
+				dragStop: this.resizeMissionList,
+				title: caption,
+				buttons:
+				[
+					{
+						text: "Create new mission",
+						click: function() {
+							open('https://missions.ingress.com/');
+						}
+					},
+					{
+						text: "Ok",
+						click: function() {
+							$(this).dialog('close');
+						}
+					},
+					{
+						text: "Missions in view",
```

Comment:
Can you explain what this button is for? I don't see any difference from removing it

Reply:
The Missions list window can either show "missions in view" or missions starting at a specific portal.

When showing missions in view, the list is automatically updated when the map is scrolled or zoomed. While showing missions starting at a specific portal, the automatic update is disabled. This button is only shown when the list shows the missions starting at a specific portal and it will return the window to the mode where it shows the missions in view.

Reply:
Showing Missions in View

![image](https://user-images.githubusercontent.com/42768755/112460179-e6968b00-8d5e-11eb-856b-5bd63ad45ea9.png)

Showing Missions at one portal

![image](https://user-images.githubusercontent.com/42768755/112460230-f44c1080-8d5e-11eb-9977-ee66edf48fd1.png)